lib/fluent/plugin/statusz.rb

```diff
@@ -12,20 +12,114 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+require 'erb'
+
 # Module for collecting diagnostic information and formatting it as an
 # HTML page to serve on the /statusz endpoint.
 module Statusz
   module_function
 
-  def response
-    # TODO(davidbtucker): Add more status information here.
-    [
-      '<html>',
-      '<body>',
-      '<h1>Status</h1>',
-      "<b>Command-line flags:</b> #{CGI.escapeHTML(ARGV.join(' '))}",
-      '</body>',
-      '</html>'
-    ].join("\n") + "\n"
+  def response(plugin)
+    uptime = Time.now - SERVER_START
+    uptime_str = format('%d hr %02d min %02d sec',
+                        uptime / 3600,
+                        (uptime / 60) % 60,
+                        uptime % 60)
+    ERB.new(STATUSZ_TMPL).result(binding)
   end
 end
+
+SERVER_START = Time.now
+
+# Note: The table below doesn't include the following config params
+# because they are deprecated: auth_method, private_key_email,
+# private_key_passphrase, private_key_path.
+
+# rubocop:disable LineLength
+STATUSZ_TMPL = %(\
+<!DOCTYPE html>
+<html>
+  <head>
+    <title>Status for <%= File.basename($PROGRAM_NAME) %></title>
+    <style>
+      body {
+        font-family: sans-serif;
+      }
+      h1 {
+        clear: both;
+        width: 100%;
+        text-align: center;
+        font-size: 120%;
+        background: #eef;
+      }
+      .lefthand {
+        float: left;
+        width: 80%;
+      }
+      .righthand {
+        text-align: right;
+      }
+      td, th {
+        background-color: rgba(0, 0, 0, 0.05);
+      }
+      th {
+        text-align: left;
+      }
+    </style>
+  </head>
+
+  <body>
+    <h1>Status for <%= File.basename($PROGRAM_NAME) %></h1>
+
+    <div>
+      <div class="lefthand">
+        Started: <%= SERVER_START %><br>
+        Up <%= uptime_str %><br>
+      </div>
+    </div>
+
+    <h1>Parsed configuration</h1>
+
+    <table>
+      <tr><th>adjust_invalid_timestamps</th><td><%= plugin.adjust_invalid_timestamps %></td></tr>
```

Comment:
This seems like something you can generate programmatically... Making it a hand-typed constant is too error-prone...

Reply:
I looked into it, but I didn't see any better way.

Reply:
Ah nice, thanks!

Reply:
Done, thanks!